### DISCUSSION TITLE
After graphscope session is closed,I create session again and appears following erro.

### DISCUSSION THREAD

**Discussion by neblen** [alibaba/GraphScope#381] [Q&A] [answered]
2021-06-07 06:18:26,364 [INFO][session:452]: Connecting graphscope session with address: xx.xx.xx.xx:xx
2021-06-07 06:18:26,370 [INFO][rpc:110]: GraphScope coordinator service connected.
---------------------------------------------------------------------------
ConnectionError                           Traceback (most recent call last)
<ipython-input-8-06c29e0af468> in <module>
      2 graphscope.set_option(show_log=True)
      3 graphscope.set_option(log_level="DEBUG")
----> 4 sess = graphscope.session(addr='xx.xx.xx.xx:xx', k8s_service_type="LoadBalancer")
      5 g = sess.g()
      6 g = g.add_vertices("/tmp/testingdata/ogbn_mag_small/paper.csv", label="paper")

/opt/conda/lib/python3.8/site-packages/graphscope/client/utils.py in wrapper(*args, **kwargs)
    154             func.__defaults__ = tuple(new_defaults)
    155 
--> 156             return_value = func(*args, **kwargs)
    157 
    158             # Restore original defaults.

/opt/conda/lib/python3.8/site-packages/graphscope/client/session.py in __init__(self, config, cluster_type, addr, num_workers, preemptive, k8s_namespace, k8s_service_type, k8s_gs_image, k8s_etcd_image, k8s_gie_graph_manager_image, k8s_zookeeper_image, k8s_image_pull_policy, k8s_image_pull_secrets, k8s_coordinator_cpu, k8s_coordinator_mem, k8s_etcd_num_pods, k8s_etcd_cpu, k8s_etcd_mem, k8s_zookeeper_cpu, k8s_zookeeper_mem, k8s_gie_graph_manager_cpu, k8s_gie_graph_manager_mem, k8s_vineyard_daemonset, k8s_vineyard_cpu, k8s_vineyard_mem, vineyard_shared_mem, k8s_engine_cpu, k8s_engine_mem, k8s_mars_worker_cpu, k8s_mars_worker_mem, k8s_mars_scheduler_cpu, k8s_mars_scheduler_mem, k8s_volumes, k8s_waiting_for_delete, timeout_seconds, dangling_timeout_seconds, with_mars, **kw)
    489         # create and connect session
    490         with CaptureKeyboardInterrupt(self.close):
--> 491             self._connect()
    492 
    493         self._disconnected = False

/opt/conda/lib/python3.8/site-packages/graphscope/client/session.py in _connect(self)
    844                 self._config_params["num_workers"],
    845                 self._config_params["k8s_namespace"],
--> 846             ) = self._grpc_client.connect(
    847                 cleanup_instance=not bool(self._config_params["addr"]),
    848                 dangling_timeout_seconds=self._config_params[

/opt/conda/lib/python3.8/site-packages/graphscope/client/rpc.py in connect(self, cleanup_instance, dangling_timeout_seconds)
    119 
    120     def connect(self, cleanup_instance=True, dangling_timeout_seconds=60):
--> 121         return self._connect_session_impl(
    122             cleanup_instance=cleanup_instance,
    123             dangling_timeout_seconds=dangling_timeout_seconds,

/opt/conda/lib/python3.8/site-packages/graphscope/client/rpc.py in with_grpc_catch(*args, **kwargs)
     44     def with_grpc_catch(*args, **kwargs):
     45         try:
---> 46             return fn(*args, **kwargs)
     47         except grpc.RpcError as exc:
     48             if isinstance(exc, grpc.Call):

/opt/conda/lib/python3.8/site-packages/graphscope/client/rpc.py in _connect_session_impl(self, cleanup_instance, dangling_timeout_seconds)
    216 
    217         response = self._stub.ConnectSession(request)
--> 218         response = check_grpc_response(response)
    219 
    220         self._session_id = response.session_id

/opt/conda/lib/python3.8/site-packages/graphscope/framework/errors.py in check_grpc_response(response)
    169     error_type = _gs_error_types.get(status.code)
    170     if error_type:
--> 171         raise error_type(status.error_msg, detail)
    172     else:
    173         raise RuntimeError(

ConnectionError: 'Cannot setup more than one connection at the same time.'

**Comment by neblen** [comment]
再创建session会失败，目前的解决方式是在k8s中删掉graphscope,再通过helm重新部署
There is a method to solve this erro,which is creating graphscope again by helm.

**Comment by acezen** [comment]
it seems the previous session not closed successfully. @lidongze0629 please take a look to this question.

**Comment by lidongze0629** [comment, marked as answer]
Hi @neblen, Follow [this doc](https://artifacthub.io/packages/helm/graphscope/graphscope) to deploy GraphScope with Helm, 
you can (re)connect session with `addr` in remote mode.

![image](https://user-images.githubusercontent.com/23491543/120977881-6fda2d00-c7a6-11eb-8f3d-067d64ab2875.png)

**Reply by neblen** [reply]
ok， thank you so much. I  try it again.
